src/xdp-utils.c

```diff
@@ -278,6 +280,25 @@ xdp_app_info_get_id (XdpAppInfo *app_info)
   return app_info->id;
 }
 
+const char *
+xdp_app_info_get_desktop_id (XdpAppInfo *app_info)
+{
+  g_return_val_if_fail (app_info != NULL, NULL);
+
+  switch (app_info->kind)
+    {
+    case XDP_APP_INFO_KIND_FLATPAK:
+      return app_info->id;
```

Comment:
It might be safer to remove the `XDP_APP_INFO_KIND_FLATPAK` section and use the default behaviour of NULL. This will ensure the current flatpak behaviour continues regardless of the desktop file name.